### PR TITLE
Setup amazon s3 and create profile edit route, controller, ui and helper function

### PR DESCRIPTION
**Pull request by kevin-06-huang**
### What this PR does (required):
- allow for uploading of images and multiple images.
- sign up to Amazon S3 and set it up
- share api key on Slack
- create a route for image upload
- save to s3 and retrieve the url to save onto database
- create UI for uploading profile photo under profile settings
- setup Amazon S3 for profile picture uploading. create the proper route for handling a user uploading their profile photo and integrate this in the front-end.

### Screenshots / Videos (front-end only):
Screen Shot 2022-01-28 at 8 11 25 AM

### Any information needed to test this feature (required):
- set up .env variable in server (check Slack)

### Any issues with the current functionality (optional):
- the ability to upload multiple images is implemented but currently disabled; database needs to be modified, currently if functionality is enabled, multiple images are saved to S3 but only the last image url is saved to database

### CODE REVIEW

File: client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+const ProfilePhoto: React.FC<ProfilePhotoProps> = ({ header, currentUser, currentProfile }) => {
+  const classes = useStyles();
+  const { updateSnackBarMessage } = useSnackBar();
+
+  const [file, setFile] = useState();
+  const [images, setImages] = useState([]);
+  const [isSubmitting, setSubmitting] = useState(false);
+
+  console.log(currentProfile);
```
Comment (ezeilo-su) [inline client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx:36]: Please watch for unnecessary logs to the console. Going forward, we should be cleaning up our logs done during development. Thanks!

File: client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+  const fileSelected = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const files = event.target.files!;
+    //setFile(file);
```
Comment (ezeilo-su) [inline client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx:40]: Let's get rid of this comment

File: client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    <Box
+      sx={{
+        width: 600,
+        margin: '0 auto',
+      }}
+    >
+      <SettingHeader header={header} />
+      <Box textAlign="center" marginTop={5}>
+        <form>
+          <input
```
Comment (ezeilo-su) [inline client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx:79]: Let's use the MUI `InputLabel` for this

File: server/controllers/image.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const asyncHandler = require("express-async-handler");
+
+const uploadFile = require("../utils/s3");
+
+// @route POST /users
+// @desc Search for users
+// @access Private
```
Comment (ezeilo-su) [inline server/controllers/image.js:7]: Good documentation

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Hey @kevin-06-huang! Good work with this one. Just a few comments to address.
Happy coding!

File: client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx (hunk 1 above)
Comment (kevin-06-huang) [reply client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx:36]: Understood, addressed.

File: client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx (hunk 2 above)
Comment (kevin-06-huang) [reply client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx:40]: Understood, removed.

File: client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx (hunk 3 above)
Comment (kevin-06-huang) [reply client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx:79]: Understood, using the MUI Input instead (you mean Input right?)

File: server/controllers/image.js (hunk 4 above)
Comment (kevin-06-huang) [reply server/controllers/image.js:7]: Noted, my bad lol; changed to better reflect method.

**Comment by rajivtitus** [comment]
@kevin-06-huang Please wait for PR approvals before merging :)

File: client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  const [isSubmitting, setSubmitting] = useState(false);
+
+  const fileSelected = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const files = event.target.files!;
+    setSubmitting(true);
+    const promises = [];
+    for (let i = 0; i < files.length; i++) {
+      promises.push(postPhoto(files[i]));
+      Promise.all(promises)
+        .then((dataArray) => {
```
Comment (rajivtitus) [inline client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx:44]: Let's also chain a `catch` block to handle any promise rejections

File: server/utils/s3.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+const secretAccessKey = process.env.AWS_SECRET_KEY
+
+const s3 = new S3({
+    region,
+    accessKeyId,
+    secretAccessKey
+});
+
+const uploadFile = (file) => {
+    const fileStream = fs.createReadStream(file.path)
```
Comment (rajivtitus) [inline server/utils/s3.js:15]: Let's also perform file type validation to only accept image type files

**Review by rajivtitus — COMMENTED** [review]
Good work with this 👍🏽 
Just a couple of comments for you to have a look

**Comment by kevin-06-huang** [comment]
Sorry I just saw the comments I'll deal with them right now

File: client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx (hunk 5 above)
Comment (kevin-06-huang) [reply client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx:44]: Something like ` .catch((error) => {
        console.log(error);
      });`

File: server/utils/s3.js (hunk 6 above)
Comment (kevin-06-huang) [reply server/utils/s3.js:15]: Addressed in newest commit at image controller in server

**Comment by kevin-06-huang** [comment]
My apologies for merging prematurely, I thought this was approved.

**Pull request merged**